modules/publicize/publicize.php

```diff
@@ -119,7 +119,7 @@ function __construct() {
 
 		add_action( 'init', array( $this, 'add_post_type_support' ) );
 		add_action( 'init', array( $this, 'register_post_meta' ), 20 );
-		add_action( 'init', array( $this, 'register_gutenberg_extension' ), 30 );
+		add_action( 'rest_api_init', array( $this, 'register_gutenberg_extension' ), 30 );
```

Comment:
This way we won't have the extension registered for non-REST-API requests. Are we sure that's what we want, and that this won't be useful outside of the REST API context?

Reply:
I'll revert back to `init`, since I've just verified that that still works. (I think I made the change at some point into the process when I still had some of `Jetpack_Gutenberg`'s own initialization depend too much on that hook and it didn't work.)

Some more general notes on the question:

I've also found myself wondering about this while working on this PR (and looking for appropriate actions to hook into). For most cases, we currently really only seem to need block registration and availability status for the `/gutenberg/available-extensions` endpoint (and not much otherwise). There's at least one counter-example I know of: If the contact form's child blocks aren't registered on the server side, they will show their default content on the frontend instead of whatever changes the user has made to them. (This was the bug caused by #11075 that required me to revert it.)

If we also start using Gutenberg's own [mechanism](https://github.com/WordPress/gutenberg/blob/95edac1e42cb10ed7da9f406696cdc85d6e476d5/lib/client-assets.php#L826-L833) to enqueue frontend-side block assets (which I've discovered only recently) instead of `Jetpack_Gutenberg`s (which I think would make total sense), server/side block registration will also be relevant for that.

Reply:
I think the case I was most worried about here (and that prompted me to change the action to hook into) was that previously, we were using block registration via a callback here (which was one of `Jetpack_Gutenberg`'s previous workarounds for problems with initialization order). Specifically, we want to set Publicize availability to `false` (because `unauthorized`) for users that don't have sufficient capabilities. 

I just changed back the hook to `init` locally and tested with a user with `contributor` role, and Publicize is indeed not showing up (and `unavailable_reason` set to `unauthorized`). I'll commit the change then :smile:

Reply:
dc2d006